decode.go

```diff
@@ -625,7 +635,17 @@ func (d *decoder) mappingStruct(n *node, out reflect.Value) (good bool) {
 		if !d.unmarshal(ni, name) {
 			continue
 		}
-		if info, ok := sinfo.FieldsMap[name.String()]; ok {
+		ns := name.String()
+		if d.strict {
+			for _, df := range doneFields {
+				if ns == df {
+					d.terrors = append(d.terrors, fmt.Sprintf("line %d: field %s already set in struct %s", n.line+1, ns, out.Type()))
```

Comment:
perhaps `set in type %s` so that we don't stutter when the type is an unnamed struct?

Reply:
Fixed.